### PR TITLE
Chore/hl testnet deploy

### PR DESCRIPTION
**Pull request by auditsbydanny**

### CODE REVIEW

File: contracts/src/scripts/DeployLiquity2.s.sol [hunk 1]
```diff
@@ -216,53 +221,74 @@ contract DeployLiquity2Script is Script, StdCheats, MetadataDeployment {
         gasCompZapperImpl = new GasCompZapper();
         troveNFTImpl = new TroveNFT();
         metadataNFTImpl = new MetadataNFT();
+        hlPriceFeedImpl = new HLPriceFeed();
 
         TroveManagerParams[] memory troveManagerParamsArray = new TroveManagerParams[](2);
 
+        /// @dev TODO: check these params for HL mainnet
         troveManagerParamsArray[0] = TroveManagerParams(150e16, 110e16, 110e16, 5e16, 10e16); // WETH
```
Comment (0x73696d616f) [inline contracts/src/scripts/DeployLiquity2.s.sol]: Let's place these in the constants

File: contracts/src/scripts/DeployLiquity2.s.sol [hunk 2]
```diff
@@ -216,53 +221,74 @@ contract DeployLiquity2Script is Script, StdCheats, MetadataDeployment {
...
-            demoTroves[8] = DemoTroveParams(1, demoAccounts[0], 0, 21e18, 2000e18, 3.3e16);
-            demoTroves[9] = DemoTroveParams(1, demoAccounts[1], 1, 16e18, 2000e18, 4.1e16);
-            demoTroves[10] = DemoTroveParams(1, demoAccounts[2], 1, 18e18, 2300e18, 3.8e16);
-            demoTroves[11] = DemoTroveParams(1, demoAccounts[3], 1, 22e18, 2200e18, 4.3e16);
-
-            demoTroves[12] = DemoTroveParams(1, demoAccounts[4], 1, 85e18, 12000e18, 7.0e16);
-            demoTroves[13] = DemoTroveParams(1, demoAccounts[5], 1, 87e18, 4000e18, 4.4e16);
-            demoTroves[14] = DemoTroveParams(1, demoAccounts[6], 1, 71e18, 11000e18, 3.3e16);
-            demoTroves[15] = DemoTroveParams(1, demoAccounts[7], 1, 84e18, 12800e18, 4.4e16);
+        if (block.chainid != 0 && vm.envOr("OPEN_DEMO_TROVES", false)) {
```
Comment (0x73696d616f) [inline contracts/src/scripts/DeployLiquity2.s.sol]: You can return early here instead, looks better.
`If (block.chainid == 0 || !vm.envOr("OPEN_DEMO_TROVES", false) return;`

File: contracts/src/scripts/DeployLiquity2.s.sol [hunk 3]
```diff
@@ -216,53 +221,74 @@ contract DeployLiquity2Script is Script, StdCheats, MetadataDeployment {
...
         troveManagerParamsArray[1] = TroveManagerParams(150e16, 120e16, 110e16, 5e16, 10e16); // stETH
 
-        // used for gas compensation and as collateral of the first branch
-        IWETH WETH = new WETHTester({_tapAmount: 100 ether, _tapPeriod: 1 days});
+        IWETH WETH;
+        /// @notice chainid == 0 is a placeholder for the mainnet chainid
+        /// @dev TODO: Update for HL mainnet
+        if (block.chainid == 0) {
+            /// @dev TODO: Change to WETH for HL mainnet
+            WETH = IWETH(0x1BCDCe8101EE46d960BF0A1E55548542BD4008D1);
```
Comment (0x73696d616f) [inline contracts/src/scripts/DeployLiquity2.s.sol]: Move to constant files

File: contracts/src/scripts/DeployLiquity2.s.sol [hunk 4]
```diff
@@ -367,14 +393,20 @@ contract DeployLiquity2Script is Script, StdCheats, MetadataDeployment {
...
-            vars.collaterals[vars.i] = new ERC20Faucet(
-                string.concat("Staked ETH", string(abi.encode(vars.i))), // _name
-                string.concat("stETH", string(abi.encode(vars.i))), // _symbol
-                100 ether, //     _tapAmount
-                1 days //         _tapPeriod
-            );
+        /// @dev TODO: Update for HL mainnet
+        if (block.chainid == 0) {
+            /// @dev TODO: Change to RETH for HL mainnet
+            vars.collaterals[1] = IERC20Metadata(0xae78736Cd615f374D3085123A210448E74Fc6393); // RETH
```
Comment (0x73696d616f) [inline contracts/src/scripts/DeployLiquity2.s.sol]: Move to constants files

File: contracts/src/scripts/DeployLiquity2.s.sol [hunk 5]
```diff
@@ -216,53 +221,74 @@ contract DeployLiquity2Script is Script, StdCheats, MetadataDeployment {
...
+                demoTroves[1] = DemoTroveParams(0, demoAccounts[1], 0, 37e18, 2400e18, 4.7e16);
+                demoTroves[2] = DemoTroveParams(0, demoAccounts[2], 0, 30e18, 4000e18, 3.3e16);
+                demoTroves[3] = DemoTroveParams(0, demoAccounts[3], 0, 65e18, 6000e18, 4.3e16);
+
+                demoTroves[4] = DemoTroveParams(0, demoAccounts[4], 0, 19e18, 2280e18, 5.0e16);
+                demoTroves[5] = DemoTroveParams(0, demoAccounts[5], 0, 48.37e18, 4400e18, 4.7e16);
+                demoTroves[6] = DemoTroveParams(0, demoAccounts[6], 0, 33.92e18, 5500e18, 3.8e16);
+                demoTroves[7] = DemoTroveParams(0, demoAccounts[7], 0, 47.2e18, 6000e18, 4.3e16);
+
+                demoTroves[0] = DemoTroveParams(1, demoAccounts[0], 0, 21e18, 2000e18, 3.3e16);
```
Comment (0x73696d616f) [inline contracts/src/scripts/DeployLiquity2.s.sol]: Why is this 0? should be 8 right?

File: contracts/src/scripts/DeployLiquity2.s.sol [hunk 6]
```diff
@@ -40,6 +41,9 @@ contract DeployLiquity2Script is Script, StdCheats, MetadataDeployment {
     using Strings for *;
     using StringFormatting for *;
 
+    uint256 constant HL_ETH_FEED_L1_INDEX = 4;
```
Comment (0x73696d616f) [inline contracts/src/scripts/DeployLiquity2.s.sol]: Move to constants file

File: contracts/src/scripts/DeployLiquity2.s.sol [hunk 7]
```diff
@@ -216,53 +221,74 @@ contract DeployLiquity2Script is Script, StdCheats, MetadataDeployment {
...
+        /// @dev TODO: check these params for HL mainnet
         troveManagerParamsArray[0] = TroveManagerParams(150e16, 110e16, 110e16, 5e16, 10e16); // WETH
         troveManagerParamsArray[1] = TroveManagerParams(150e16, 120e16, 110e16, 5e16, 10e16); // stETH
 
-        // used for gas compensation and as collateral of the first branch
-        IWETH WETH = new WETHTester({_tapAmount: 100 ether, _tapPeriod: 1 days});
+        IWETH WETH;
+        /// @notice chainid == 0 is a placeholder for the mainnet chainid
+        /// @dev TODO: Update for HL mainnet
+        if (block.chainid == 0) {
```
Comment (0x73696d616f) [inline contracts/src/scripts/DeployLiquity2.s.sol]: Let's place the mainnet chain id in the constants file

File: contracts/src/scripts/DeployLiquity2.s.sol [hunk 8]
```diff
@@ -477,7 +509,24 @@ contract DeployLiquity2Script is Script, StdCheats, MetadataDeployment {
             ))
         );
 
-        contracts.priceFeed = new PriceFeedTestnet();
+        /// @notice 998: Hyperliquid EVM Testnet
+        /// @dev TODO: Update for HL mainnet too
+        if (block.chainid == 998) {
```
Comment (0x73696d616f) [inline contracts/src/scripts/DeployLiquity2.s.sol]: Let's not leave TODO's, think this should have a flag too, `USE_MOCK_ORACLE` or smth.
If the chain id is mainnet, or `USE_MOCK_ORACLE` is false, use the hl oracle
else use the mock oracle

**Review by 0x73696d616f — CHANGES_REQUESTED** [review]
Some extra stuff besides the comments:
1. Let's make the BoldToken upgradeble and adjust the script.
2. Let's also make the mockInterestRouter upgradeable, so we can update it later.
3. And also make a setter for the `SP_YIELD_SPLIT` because we might change it. It's easier than making an upgrade just to change a var we know will change.

File: contracts/src/scripts/DeployLiquity2.s.sol (hunk 8 above)
Comment (auditsbydanny) [reply contracts/src/scripts/DeployLiquity2.s.sol]: Resolved in [2a74157](https://github.com/threesigmaxyz/bold/pull/3/commits/2a74157125ba6563ffba9ba2cb4cb162ac1fc8fc)

File: contracts/src/scripts/DeployLiquity2.s.sol (hunk 7 above)
Comment (auditsbydanny) [reply contracts/src/scripts/DeployLiquity2.s.sol]: [2a74157](https://github.com/threesigmaxyz/bold/pull/3/commits/2a74157125ba6563ffba9ba2cb4cb162ac1fc8fc)

File: contracts/src/scripts/DeployLiquity2.s.sol (hunk 5 above)
Comment (auditsbydanny) [reply contracts/src/scripts/DeployLiquity2.s.sol]: Resolved in [2a74157](https://github.com/threesigmaxyz/bold/pull/3/commits/2a74157125ba6563ffba9ba2cb4cb162ac1fc8fc)

File: contracts/src/scripts/DeployLiquity2.s.sol (hunk 4 above)
Comment (auditsbydanny) [reply contracts/src/scripts/DeployLiquity2.s.sol]: Resolved in [2a74157](https://github.com/threesigmaxyz/bold/pull/3/commits/2a74157125ba6563ffba9ba2cb4cb162ac1fc8fc)

File: contracts/src/scripts/DeployLiquity2.s.sol (hunk 3 above)
Comment (auditsbydanny) [reply contracts/src/scripts/DeployLiquity2.s.sol]: Resolved in [2a74157](https://github.com/threesigmaxyz/bold/pull/3/commits/2a74157125ba6563ffba9ba2cb4cb162ac1fc8fc)

File: contracts/src/scripts/DeployLiquity2.s.sol (hunk 2 above)
Comment (auditsbydanny) [reply contracts/src/scripts/DeployLiquity2.s.sol]: Resolved in [2a74157](https://github.com/threesigmaxyz/bold/pull/3/commits/2a74157125ba6563ffba9ba2cb4cb162ac1fc8fc)

File: contracts/src/scripts/DeployLiquity2.s.sol (hunk 1 above)
Comment (auditsbydanny) [reply contracts/src/scripts/DeployLiquity2.s.sol]: Resolved in [2a74157](https://github.com/threesigmaxyz/bold/pull/3/commits/2a74157125ba6563ffba9ba2cb4cb162ac1fc8fc)

File: contracts/src/scripts/DeployLiquity2.s.sol (hunk 6 above)
Comment (auditsbydanny) [reply contracts/src/scripts/DeployLiquity2.s.sol]: Resolved in [6719602](https://github.com/threesigmaxyz/bold/pull/3/commits/6719602a88eaefd363ea0fa6ae72c55ec825f861)

**Pull request merged**